nodejs/wrapper.go

```diff
@@ -0,0 +1,299 @@
+	nc.handleNodemon()
+
+	return nc.exec(stdin, stdout, stderr)
+}
+
+// unwrap looks for the real node executable (not this wrapper).
+func (nc *nodeContext) unwrap() bool {
+	if nc == nil {
+		return false
+	}
+
+	// Here we try to find the original program.  When a program is
+	// resolved from the PATH, most shells will set argv[0] to the
+	// command and so it won't appear to exist and so the first file
+	// resolved in the PATH should be this program.
+	origInfo, err := os.Stat(nc.program)
+	origFound := err == nil
+	if err != nil && !os.IsNotExist(err) {
+		logrus.Errorf("unable to stat %q: %v", nc.program, err)
+		return false
+	}
+
+	path := nc.env["PATH"]
+	base := filepath.Base(nc.program)
+	for _, dir := range strings.Split(path, string(os.PathListSeparator)) {
+		p := filepath.Join(dir, base)
+		if pInfo, err := os.Stat(p); err == nil {
+			if !origFound {
+				// the original nc.program was not resolved, meaning this
+				// it had been resolved in the PATH, so treat this first
+				// instance as the original file and continue searching
+				logrus.Debugf("unwrap: presumed wrapper at %s", p)
+				origInfo = pInfo
+				origFound = true
+			} else if !os.SameFile(origInfo, pInfo) {
+				logrus.Debugf("unwrap: replacing %s -> %s", nc.program, p)
+				nc.program = p
+				return true
+			}
+		}
+	}
+	logrus.Errorf("unable to unwrap %q: not in PATH", base)
+	return false
+}
+
+// stripInspectArgs removes all `--inspect*` args from both the command-line and from
+// NODE_OPTIONS.  It returns the last inspect arg or "" if there were no inspect arguments.
+func (nc *nodeContext) stripInspectArgs() string {
+	foundOption := ""
+	if options, found := nc.env["NODE_OPTIONS"]; found {
+		if args, err := shell.Split(options); err != nil {
+			logrus.Warnf("NODE_OPTIONS cannot be split: %v", err)
+		} else {
+			args, inspectArg := stripInspectArg(args)
+			if inspectArg != "" {
+				logrus.Debugf("Found %q in NODE_OPTIONS", inspectArg)
+				nc.env["NODE_OPTIONS"] = shell.Join(args...)
+				foundOption = inspectArg
+			}
+		}
+	}
+	strippedArgs, inspectArg := stripInspectArg(nc.args)
+	if inspectArg != "" {
+		logrus.Debugf("Found %q in command-line", inspectArg)
+		nc.args = strippedArgs
+		foundOption = inspectArg
+	}
+	return foundOption
+}
+
+func (nc *nodeContext) handleNodemon() {
+	if nodeDebug, found := nc.env["NODE_DEBUG"]; found {
+		// look for the nodemon script (if it appears) and insert the --inspect argument
+		for i, arg := range nc.args {
+			if len(arg) > 0 && arg[0] != '-' && strings.Contains(arg, "/nodemon") {
+				nc.args = append(nc.args, "")
+				copy(nc.args[i+2:], nc.args[i+1:])
+				nc.args[i+1] = nodeDebug
+				delete(nc.env, "NODE_DEBUG")
+				logrus.Debugf("special handling for nodemon: %q", nc.args)
+				return
+			}
+		}
+	}
+}
+
+func (nc *nodeContext) addNodeArg(nodeArg string) {
+	// find the script location and insert the provided argument
+	if len(nc.args) == 0 {
```

Comment:
Can those 4 lines be removed? Seems to handle a case that's already handled by the rest of the code